### PR TITLE
feat: fix api  metrics

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
新增api metrics middleware

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Comment by wklken** [comment]
lint挂了

File: src/core-api/pkg/middleware/metrics.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+		duration := time.Since(start)
+		status := strconv.Itoa(c.Writer.Status())
+		// request count
+		metric.RequestCount.With(prometheus.Labels{
+			"method": c.Request.Method,
+			"path":   c.Request.URL.Path,
+			"status": status,
+		}).Inc()
+
```
Comment (wklken) [inline src/core-api/pkg/middleware/metrics.go:47]: path 有办法变更 `pattern`吗, 即忽略具体参数; 
或者, 改变events的 url
----
`/:micro_gateway_instance_id/release/:publish_id/events/`
原因: publish_id是无限递增的, 会导致metrics非常大

Comment (wklken) [reply src/core-api/pkg/middleware/metrics.go:47]: apisix 中使用的 matched_uri, 替代了path

**Review by alex-smile — APPROVED** [review]

File: src/core-api/pkg/middleware/metrics.go (hunk 1 above)
Comment (alex-smile) [reply src/core-api/pkg/middleware/metrics.go:47]: /:micro_gateway_instance_id/release/:publish_id/events/

现在的写法，将导致 key 数量特别大？

Comment (Han-Ya-Jun) [reply src/core-api/pkg/middleware/metrics.go:45]: 已改成matched_uri

**Review by wklken — APPROVED** [review]

**Pull request merged**